File: Second year/MPP/C#/Incercare/service/Service.cs
using System;
using System.Collections.Generic;
using System.Configuration;
using System.Xml.Schema;
using Incercare.domain;
using Incercare.repository;

namespace Incercare.service
{
    public class Service
    {
        private UserRepository repou;
        private ZborRepository repoz;
        private BiletRepository repob;
        private BiletTuristRepository repobt;
        private int user_id;

        
        static string GetConnectionStringByName(string name)
        {
            string returnValue = null;

            ConnectionStringSettings settings = ConfigurationManager.ConnectionStrings[name];

            if (settings != null)
                returnValue = settings.ConnectionString;

            return returnValue;
        }
        
        public Service(IDictionary<String,String> props)
        {
            log4net.Config.XmlConfigurator.Configure();

            this.repou = new UserRepository(props);
            this.repoz = new ZborRepository(props);
            this.repob = new BiletRepository(props);
            this.repobt = new BiletTuristRepository(props);
        }

        public List<Zbor> findAllFlights()
        {
            return repoz.findAll();
        }

        public User login(string username, string password)
        {
            List<User> users = repou.findAll();
            foreach(User user in users)
            {
                if (user.username == username)
                {
                    user_id = user.id;
                    return user;
                }
            }

            return null;
        }

        public List<Zbor> findFlight(string destination, DateTime data)
        {
            List<Zbor> zboruri = repoz.findAll();
            List<Zbor> newzboruri = new List<Zbor>();
            foreach (Zbor zbor in zboruri)
            {
                if (zbor.destinatie==destination && zbor.data_ora.Date==data.Date)
                    newzboruri.Add(zbor);
            }

            return newzboruri;
        }

        public int generate_id()
        { 
            List<Bilet> bilete = repob.findAll();
            int maxi = 0;
            foreach (Bilet bilet in bilete)
            {
                if (bilet.id > maxi)
                    maxi = bilet.id;
            }

            maxi = maxi + 1;
            return maxi;
        }


        public Zbor findOne(int id)
        {
            foreach (Zbor zbor in repoz.findAll())
            {
                if (zbor.id == id)
                    return zbor;
            }

            return null;
        }
        public void buyTicket(int zbor_id, string client_nume, string client_adresa, int nr_locuri, List<String> turisti)
        {
            int id = generate_id();
            Bilet bilet = new Bilet(id, zbor_id, user_id, client_nume, client_adresa, nr_locuri);
            repob.save(bilet);
            Zbor zbor = findOne(zbor_id);
            int locuri = zbor.locuri;
            zbor.locuri = locuri - nr_locuri;
            repoz.update(zbor);
          ///  foreach(string turist in turisti)
           /// {
           ///     BiletTurist bt = new BiletTurist(id, turist);
             ///   repobt.save(bt);
            ///}
        }

        public void logout()
        {
            user_id = -1;
        }

        public List<Zbor> getAvailableFlights()
        {
            List<Zbor> zboruri = new List<Zbor>();
            foreach (Zbor zbor in repoz.findAll())
            {
                if (zbor.locuri>0)
                    zboruri.Add(zbor);
            }

            return zboruri;
        }
    }
}

File: Second year/MPP/Java_MPP_Bun/src/main/java/com/example/java_mpp_bun/service/Service.java
package com.example.java_mpp_bun.service;



import com.example.java_mpp_bun.domain.Bilet;
import com.example.java_mpp_bun.domain.User;
import com.example.java_mpp_bun.domain.Zbor;
import com.example.java_mpp_bun.repository.BiletRepository;
import com.example.java_mpp_bun.repository.UserRepository;
import com.example.java_mpp_bun.repository.ZborRepository;
import com.example.java_mpp_bun.utils.ObservableImplementat;

import java.io.FileReader;
import java.io.IOException;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.Objects;
import java.util.Properties;


public class Service extends ObservableImplementat {
    UserRepository repou;
    ZborRepository repoz;
    BiletRepository repob;

    Integer user_id;

    public void setService(){
        Properties props = new Properties();
        try {
            props.load(new FileReader("C:\\Documente\\GitHub\\mpp-proiect-java-Andra1028\\Project1_MPP\\bd.config"));
        } catch (
                IOException e) {
            System.out.println("Cannot find bd.config " + e);
        }

        repou = new UserRepository(props);
        repoz = new ZborRepository(props);
        repob = new BiletRepository(props);
    }

    public List<Zbor> findAllFlignts()
    {
        return repoz.findAll();
    }

    public User login(String username, String password){
        User user= repou.findByName(username);
        if(user==null)
            throw new RuntimeException("User not found");
        user_id=user.getID();
        return user;
    }

    public List<Zbor> findFlight(String destination, LocalDate data){
        List<Zbor> zboruri = new ArrayList<>();
        for(Zbor zbor: repoz.findAll()){
            if(zbor.getLocuri() >0 && Objects.equals(zbor.getDestinatie(), destination)
                    && zbor.getData_ora().getYear()==data.getYear() && zbor.getData_ora().getMonth()==data.getMonth()
                    && zbor.getData_ora().getDayOfMonth()==data.getDayOfMonth())
                zboruri.add(zbor);
        }
        return zboruri;
    }

    public int generateId(){
        int maxi =0;
        for(Bilet ticket: repob.findAll())
        {
            if(ticket.getID()>maxi)
                maxi=ticket.getID();
        }
        maxi++;
        return maxi;
    }


    public void buyTicket(Integer zbor_id, String client_nume, String client_adresa, Integer nr_locuri, String turisti)
    {
        int id= generateId();
        List<String> t= List.of(turisti.split(","));
        Bilet bilet = new Bilet(zbor_id,user_id,client_nume,t,client_adresa,nr_locuri);
        bilet.setID(id);
        repob.save(bilet);
        Zbor zbor = repoz.findOne(zbor_id);
        zbor.setLocuri(zbor.getLocuri()-nr_locuri);
        repoz.update(zbor);
    }

    public void logout(){
        user_id= null;
    }

    public List<Zbor> availableFlights(){
        return null;
    }

}
